geospaas_harvesting/ingesters.py

```diff
@@ -240,31 +255,30 @@ def ingest(self, urls, *args, **kwargs):
                 db_executor.submit(self._thread_ingest_dataset)
 
             # Launch threads which fetch datasets attributes and put them in the queue
-            try:
-                with concurrent.futures.ThreadPoolExecutor(
-                        max_workers=self.max_fetcher_threads,
-                        thread_name_prefix=self.__class__.__name__ + '.attr') as attr_executor:
+            with concurrent.futures.ThreadPoolExecutor(
+                    max_workers=self.max_fetcher_threads,
+                    thread_name_prefix=self.__class__.__name__ + '.attr') as attr_executor:
+                try:
                     attr_futures = []
-                    for url in urls:
-                        if self._uri_exists(url):
-                            self.LOGGER.info("'%s' is already present in the database", url)
-                        else:
-                            attr_futures.append(attr_executor.submit(
-                                self._thread_get_normalized_attributes, url, *args, *kwargs))
-            except KeyboardInterrupt:
-                for future in attr_futures:
-                    future.cancel()
-                self.LOGGER.debug(
-                    'Cancelled future fetching threads, waiting for the running threads to finish')
-                concurrent.futures.wait(attr_futures)
-                raise
-            finally:
-                # Wait for all queue elements to be processed and stop database access threads
-                self.LOGGER.debug('Waiting for all the datasets in the queue to be ingested...')
-                self._to_ingest.join()
-                self.LOGGER.debug('Stopping all database access threads')
-                for _ in range(self.max_db_threads):
-                    self._to_ingest.put(None)
+                    for dataset_info in datasets_info:
```

Comment:
```suggestion
                    for dataset_info in datasets_infos:
```
Please use the proper name for variable and pay attention to their plural states

Reply:
It is already a plural: `datasetS_info`. ("information" is an uncountable noun in English: it can't be set in a plural form)
But this name is clearly confusing, so I renamed `datasets_info` to `datasets_to_ingest`.